arch/risc-v/src/mpfs/mpfs_dsn.c

```diff
@@ -91,9 +98,7 @@ int mpfs_read_dsn(uint8_t *dsn, size_t len)
 
   while ((getreg32(SERVICES_SR) & SCBCTRL_SERVICESSR_BUSY) && --retries > 0)
     {
-      leave_critical_section(flags);
       nxsig_usleep(1000);
```

Comment:
it's wrong to sleep while holding spinlock

Reply:
The old implementation is equally wrong then, I think we need mutex here instead?

Reply:
Or maybe just remove the loop.. the device should be free if the lock is available

Reply:
could be

Reply:
All is now fixed, thanks for the comments